src/ClientServices/WindowsProcessLauncherClient.cpp

```diff
@@ -0,0 +1,182 @@
+  grpc::Status status =
+      windows_process_launcher_service_->LaunchProcess(context.get(), request, &response);
+
+  if (!status.ok()) {
+    ORBIT_ERROR("\"LaunchProcess\" grpc call failed: code=%d, message=%s", status.error_code(),
+                status.error_message());
+    return ErrorMessage(status.error_message());
+  }
+
+  // Keep track of launched processes.
+  const orbit_grpc_protos::ProcessInfo& process_info = response.process_info();
+  LaunchedProcess launched_process;
+  launched_process.process_info_ = process_info;
+  launched_process.state_ = process_to_launch.spin_at_entry_point()
+                                ? LaunchedProcess::State::kSpinningAtEntryPoint
+                                : LaunchedProcess::State::kExecutingOrExited;
+  {
+    absl::MutexLock lock(&launched_processes_by_pid_mutex_);
+    launched_processes_by_pid_.emplace(process_info.pid(), launched_process);
+  }
+
+  return process_info;
+}
+
+ErrorMessageOr<void> WindowsProcessLauncherClientImpl::SuspendProcessSpinningAtEntryPoint(
+    uint32_t pid) {
+  ORBIT_SCOPE_FUNCTION;
+  ORBIT_CHECK(IsProcessSpinningAtEntryPoint(pid));
+  SuspendProcessSpinningAtEntryPointRequest request;
+  SuspendProcessSpinningAtEntryPointResponse response;
+  request.set_pid(pid);
+
+  std::unique_ptr<grpc::ClientContext> context = CreateContext();
+  grpc::Status status = windows_process_launcher_service_->SuspendProcessSpinningAtEntryPoint(
+      context.get(), request, &response);
+
+  if (!status.ok()) {
+    ORBIT_ERROR("\"SuspendProcessSpinningAtEntryPoint\" grpc call failed: code=%d, message=%s",
+                status.error_code(), status.error_message());
+    return ErrorMessage(status.error_message());
+  }
+
+  UpdateLaunchedProcessState(pid, LaunchedProcess::State::kSuspendedAtEntryPoint);
+  return outcome::success();
+}
+
+ErrorMessageOr<void> WindowsProcessLauncherClientImpl::ResumeProcessSuspendedAtEntryPoint(
+    uint32_t pid) {
+  ORBIT_SCOPE_FUNCTION;
+  ORBIT_CHECK(IsProcessSuspendedAtEntryPoint(pid));
+  ResumeProcessSuspendedAtEntryPointRequest request;
+  ResumeProcessSuspendedAtEntryPointResponse response;
+  request.set_pid(pid);
+
+  std::unique_ptr<grpc::ClientContext> context = CreateContext();
+  grpc::Status status = windows_process_launcher_service_->ResumeProcessSuspendedAtEntryPoint(
+      context.get(), request, &response);
+
+  if (!status.ok()) {
+    ORBIT_ERROR("\"ResumeProcessSuspendedAtEntryPoint\" grpc call failed: code=%d, message=%s",
+                status.error_code(), status.error_message());
+    return ErrorMessage(status.error_message());
+  }
+
+  UpdateLaunchedProcessState(pid, LaunchedProcess::State::kExecutingOrExited);
+  return outcome::success();
+}
+
+void WindowsProcessLauncherClientImpl::UpdateLaunchedProcessState(
+    uint32_t pid, LaunchedProcess::State new_state) {
+  absl::MutexLock lock(&launched_processes_by_pid_mutex_);
+  LaunchedProcess& launched_process = launched_processes_by_pid_.at(pid);
```

Comment:
The existence of pid in launched_processes_by_pid_ is guaranteed here, right? Maybe just add a ORBIT_CHECK.